app/scripts/lib/diagnostics-reporter.js

```diff
@@ -0,0 +1,71 @@
+class DiagnosticsReporter {
+
+  constructor ({ firstTimeInfo, version }) {
+    this.firstTimeInfo = firstTimeInfo
+    this.version = version
+  }
+
+  async reportOrphans(orphans) {
+    try {
+      await this.submit({
```

Comment:
You should be able to return `this.submit(...)` directly and this function doesn't have to be marked as `async`. In either case, this will return a Promise that will resolve when the submission is complete.

Reply:
the `await` here makes sure the try/catch catches errors coming from `this.submit`, tho we are catching all errors inside `this.submit`
we could remove the error handling in `this.submit` so we know which method errored